NCM_APIv2_class/v2_example_class.py

```diff
@@ -0,0 +1,137 @@
+        'X-CP-API-ID': '',
+        'X-CP-API-KEY': '',
+        'Accept': '*/*',
+        'Content-Type': 'application/json',
+    }
+
+    def get_data(self, url, json_output):
+
+        resp_data = {'data': []}
+
+        while url:
+
+            with requests.Session() as s:
+
+                retries = Retry(total=5,  # Total number of retries to allow.
+                                backoff_factor=2,
+                                status_forcelist=[408,  # 408 Request Timeout
+                                                  500,  # 500 Internal Server Error
+                                                  502,  # 502 Bad Gateway
+                                                  503,  # 503 Service Unavailable
+                                                  504,  # 504 Gateway Timeout
+                                                  ],
+                                )
+                a = requests.adapters.HTTPAdapter(max_retries=retries)
+                s.mount('https://', a)
+
+                try:
+                    logging.info(url)
+                    print(url)
+                    r = s.get(url, headers=self.headers,
+                              timeout=30, stream=True)
+
+                    if r.status_code != 200:
+                        logging.error(str(r.status_code) + ": " + str(r.text))
+                        s.get('https://accounts.cradlepointecm.com/logout')
+
+                    else:
+                        self.resp = json.loads(r.content.decode("utf-8"))
+
+                        if len(self.resp['data']) < 1:
+                            logging.warning('No response data was returned.')
+                            return self.resp
+
+                        else:
+                            for item in self.resp['data']:
+                                resp_data['data'].append(item)
+
+                            if self.resp['meta']['next']:
+                                url = self.resp['meta']['next']
+                            else:
+                                url = None
+
+                except Exception as e:
+                    logging.error("Exception:", e)
+                    raise
+
+        data = json.dumps(resp_data, indent=4, sort_keys=False)
+
+        with open(f'json/{json_output}.json', 'w') as outfile:
+            outfile.write(data)
+            outfile.close()
+
+        return data
+
+
+def create_dirs():
+
+    base_path = os.getcwd()
+
+    directorys = ['json', 'logs']
+
+    for path in directorys:
+
+        dir_path = os.path.join(base_path, path)
+
+        if not os.path.exists(dir_path):
+            try:
+                os.makedirs(dir_path)
+            except OSError:
+                print (f"Creation of the directory %s failed" % dir_path)
+            else:
+                print ("Successfully created the directory %s " % dir_path)
+        else:
+            print ("%s already exists" % dir_path)
+
+
+if __name__ == '__main__':
```

Comment:
Oh, wait, this is a program, not a module. Never mind what I said earlier. Put this under "examples" not "lib" and replace the get.py that is already there, or merge it's features into this one. And never mind what I said about "logging" vs "print", in a standalone program either is fine but pick one.